### ISSUE TITLE
References to primitive values are resolved as objects

### ISSUE THREAD

**Issue by madi-tracksuit** [json-schema-faker/json-schema-faker#801]
Referencing objects are inlined and mocked correctly, however referencing primitive types (ie: Scalars) resolve to empty objects

Spec:
```json
"CategoryId": {
  "type": "integer",
  "title": "CategoryId",
  "minimum": 0,
  "description": "An ID representing a category.\n"
},
"Category": {
  "type": "object",
  "title": "Category",
  "properties": {
    "id": { "$ref": "#/components/schemas/CategoryId" },
    "name": { "type": "string" }
  },
  "required": ["id", "name"]
},
```

Resulting mock:

```json
"CategoryId": 64175351,
"Category": { "id": {}, "name": "ullamco esse" },
```

However referencing an object that uses a reference to a value works fine

Spec:

```json
"AccountBrandData": {
  "type": "object",
  "title": "AccountBrandData",
  "properties": {
    "category": { "$ref": "#/components/schemas/Category" },
  },
},
```

Resulting mock:
```json
 "AccountBrandData": {
  "category": { "id": 19809994, "name": "velit aliqua dolor qui" },
},
```

**Comment by pateketrueke** [comment]
I'll review this asap, so much thanks for the details!

**Comment by madi-tracksuit** [comment]
Is there anything I can do to help move this along?

**Comment by pateketrueke** [comment]
> Is there anything I can do to help move this along?

well, if you want to tackle this one is up to you, however I need to make some spare time in order to get involved, may be during the next week but I am still not sure, sorry!

**Comment by madi-tracksuit** [comment]
No problems! Could you give me a pointer where you think the source of the issue could be? Don't have a huge amount of time to get up to scratch on the codebase, but if you can point me to the likely source (ref parser somewhere?) I can have a look

**Comment by madi-tracksuit** [comment]
Did some poking around, looks like the ref in the properties isn't getting resolved properly, and the reason it's output as an empty object is because the `$ref` property is deleted after it tries to resolve it.

Haven't gotten to the bottom of why the ref merge isn't working though

**Comment by madi-tracksuit** [comment]
It's definitely failed dereferencing, because using https://www.npmjs.com/package/@openapi-generator-plus/json-schema-ref-parser and passing in a resolved spec works well. Would you consider just using a well tested external lib for the spec resolution step?

**Comment by pateketrueke** [comment]
> Would you consider just using a well tested external lib for the spec resolution step?

Yes, that's the correct solution imho. I tried before, but everything breaks, if you want to hop into that we can fix a lot of related issues!

I was thinking that we could use the same approach as for generators, so we can decouple the resolver from the library itself, and then we can just register a resolver implementation.

What do you think?
